pipelines/legionPipeline.groovy

```diff
@@ -48,13 +48,15 @@ def createCluster() {
 }
 
 def createGCPCluster() {
+  withCredentials([
+  sshUserPrivateKey(credentialsId: "${env.legionCicdGitlabKey}", keyFileVariable: 'gitKey')]) {
     withCredentials([
     file(credentialsId: "${env.gcpCredential}", variable: 'gcpCredential')]) {
         withCredentials([
         file(credentialsId: "${env.param_cluster_name}-secrets", variable: 'secrets')]) {
             withAWS(credentials: 'kops') {
                 wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
-                    docker.image("${env.param_docker_repo}/k8s-terraform:${env.param_legion_infra_version}").inside("-e GOOGLE_CREDENTIALS=${gcpCredential} -e CLUSTER_NAME=${env.param_cluster_name} -u root") {
+                    docker.image("${env.param_docker_repo}/k8s-terraform:${env.param_legion_infra_version}").inside("-e GOOGLE_CREDENTIALS=${gcpCredential} -e CLUSTER_NAME=${env.param_cluster_name} -u root -v ${WORKSPACE}/legion-cicd/terraform/env_types/dns:/opt/legion/terraform/dns -v ${WORKSPACE}/legion-cicd/terraform/env_profiles/dns.tfvars:/opt/legion/terraform/env_profiles/dns.tfvars -v ${gitKey}:/root/.ssh/id_rsa ") {
```

Comment:
Why do we need to mount git ssh key here?

Reply:
because DNS terraform module stored in separate private branch. So we need to access it when run `terraform init`

Reply:
we don't need to mount the key explicitly, the checkout will work if we just specify the key in sshagent() because it's available through the var and stored in mounted workspace